src/test/java/org/candlepin/subscriptions/tally/facts/FactNormalizerTest.java

```diff
@@ -662,6 +664,36 @@ void testNullSocketsMarketplaceDefaulting() {
     assertEquals(0, normalizedFacts.getSockets());
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"x86_64", "i386", "i686"})
+  void testQpcSystemArchSetRhelForX86Product(String arch) {
+    NormalizedFacts normalized =
+        normalizer.normalize(createQpcHost("RHEL", arch, clock.now()), hypervisorData());
+    assertThat(normalized.getProducts(), Matchers.hasItem("RHEL for x86"));
+  }
+
+  @Test
+  void testQpcSystemArchSetRhelForArm() {
+    NormalizedFacts normalized =
+        normalizer.normalize(createQpcHost("RHEL", "aarch64", clock.now()), hypervisorData());
+    assertThat(normalized.getProducts(), Matchers.hasItem("RHEL for ARM"));
+  }
+
+  @Test
+  void testQpcSystemArchSetRhelForIbmPower() {
+    NormalizedFacts normalized =
+        normalizer.normalize(createQpcHost("RHEL", "ppc64le", clock.now()), hypervisorData());
+    assertThat(normalized.getProducts(), Matchers.hasItem("RHEL for IBM Power"));
+  }
+
+  @Test
+  void testQpcProductIdFromEngId() {
+    var host = createQpcHost("RHEL", "Test", clock.now());
+    host.setQpcProductIds("69");
```

Comment:
```suggestion
    host.setProducts("69");
```

Since https://github.com/RedHatInsights/rhsm-subscriptions/pull/2604, qpcProductIds no longer exists. so I think the right replacement for this test is to use `host.setProducts("69")`.